core/src/test/java/org/elasticsearch/common/network/NetworkServiceTests.java

```diff
@@ -95,36 +96,38 @@ public void testBindAnyLocalV4() throws Exception {
      */
     public void testBindAnyLocalV6() throws Exception {
         NetworkService service = new NetworkService(Settings.EMPTY);
-        assertEquals(InetAddress.getByName("::"), service.resolveBindHostAddress("::")[0]);
+        assertEquals(InetAddress.getByName("::"), service.resolveBindHostAddresses(new String[] { "::" })[0]);
     }
 
     /** 
      * ensure specifying wildcard ipv4 address selects reasonable publish address 
      */
     public void testPublishAnyLocalV4() throws Exception {
-        InetAddress expected = null;
+        InetAddress expected[] = null;
         try {
-            expected = NetworkUtils.getFirstNonLoopbackAddresses()[0];
+            expected = NetworkUtils.getFirstNonLoopbackAddresses();
         } catch (Exception e) {
             assumeNoException("test requires up-and-running non-loopback address", e);
         }
+        NetworkUtils.sortAddresses(Arrays.asList(expected));
         
         NetworkService service = new NetworkService(Settings.EMPTY);
-        assertEquals(expected, service.resolvePublishHostAddress("0.0.0.0"));
+        assertEquals(expected[0], service.resolvePublishHostAddresses(new String[] { "0.0.0.0" }));
```

Comment:
hmm, well the test needs to probably be fixed anyway. I want to kill this "first nonloopback" :)

But yeah we should test, without simply repeating the logic that is happening in the code, that does not really test anything...